test/integration/withdrawProfit.ts

```
@@ -0,0 +1,82 @@
+import chai from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+chai.use(chaiAsPromised);
+const expect = chai.expect;
+
+import _farmContract  from '../helpers/farm';
+import _initialStorageFarm from '../../migrations/initialStorage/farm';
+import _tokenContract from '../helpers/token';
+import _taquito from '../helpers/taquito';
+import _saveContractAddress from '../../helpers/saveContractAddress';
+import accounts from '../../scripts/sandbox/accounts';
+import { migrate as migrateDex } from '../../scripts/contracts/quipuswap';
+import { TezosOperationError } from '@taquito/taquito';
+import { contractErrors } from '../../helpers/constants';
+
+contract('farm contract', () => {
+    let farmContract;
+    let dexTokenContract;
+    let operation;
+    const recipient = accounts.bob.pkh;
+    
+    before(async () => {
+        console.log('Migrating DEX contract')
+        await migrateDex();
+        const dexTokenContractAddress = require('../../scripts/contracts/quipuswap/deployments/dex') || undefined;    
```

It would be nice to detect the migration failure here, instead of just falling back to || undefined